### PR TITLE
Support display marker and popup

### PR DESCRIPTION
**Pull request by brian-guru**
1. Add a marker based on input location.
2. added 2 props:
- title
- desc(description)
for displaying popup

### CODE REVIEW

**Comment by JeromeGillard** [comment]
Awesome, thanks 👍 for this PR.
I'll test that over the weekend, but could you already tell me if this change will not break  bbcodes that are already in existing posts (without your two new attributes)?

As far as I remember when I tried to introduce a new attribute to the bbcodes, old ones without it were rendering as plain text.
Maybe we'll need to introduce a new `[map-marker][]` as a workaround? 
I couldn't even make some attributes optional...

**Comment by brian-guru** [comment]
Hi @JeromeGillard,
Thanks for your comments.
I added support nullable bbcode attribute from my last commit.
with full attr: 
https://vietsuquan.com/d/1-chua-phu-xa-phuc-hoa-tu
and without additional attr:
https://vietsuquan.com/d/2-chua-quynh-loi-long-khanh-tu

File: src/Providers/BBCodesProvider.php [hunk 1]
```diff
@@ -11,12 +11,14 @@ class BBCodesProvider
...
-            '[map provider={TEXT2} style={TEXT3} zoom={TEXT4}]{TEXT1}[/map]',
+            '[map provider={TEXT2} style={TEXT3} zoom={TEXT4} title={TEXT5?} desc={TEXT6?}]{TEXT1}[/map]',
             '<div class="bbcode-map"
               data-map-location="{TEXT1}"
               data-map-provider="{TEXT2}"
               data-map-style="{TEXT3}"
-              data-map-zoom="{TEXT4}">
+              data-map-zoom="{TEXT4}"
+              data-map-title="{TEXT5?}"
+              data-map-desc="{TEXT6?}">
```
Comment (JeromeGillard) [inline src/Providers/BBCodesProvider.php]: image
The moment I activate those specific `TEXT5?` and `TEXT6?` I have got the entire site (front page included) that starts to raise that Exception, breaking all the Flarum instance.

Removing the `?` from `TEXT5?` and `TEXT6?` solves the problem.

File: js/src/forum/components/mapBBCode.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ export default function insertAtCursor(tilesProvider, style, zoom) {
     let pos = input.selectionStart
     let front = (input.value).substring(0, pos);
     let back = (input.value).substring(pos, input.value.length);
-    let middle = `[map provider=${tilesProvider??'osm'} style=${style??'street'} zoom=${zoom??13}]${app.translator.trans('jeromegillard-map.forum.text_editor.location_placeholder')}[/map]`;
+    let middle = `[map provider=${tilesProvider??'osm'} style=${style??'street'} zoom=${zoom??13} title='Place' desc='Updating...']${app.translator.trans('jeromegillard-map.forum.text_editor.location_placeholder')}[/map]`;
```
Comment (JeromeGillard) [inline js/src/forum/components/mapBBCode.js]: It seem to only work when there are no single quotes `'`:

👎 Not working: `title='Place'`
👍 Working `title=Place`

Hence, how to provide complex values such as `desc=Hello quoted "world"` that could contain special chars like quotes `"` or even spaces ` `?

That is the main reason I need the location to be within the bbcode `[map] free text here[/map]`.  
Is there another way to support special chars within attributes of a bbcode?

File: src/Providers/BBCodesProvider.php (hunk 1 above)
Comment (JeromeGillard) [reply src/Providers/BBCodesProvider.php]: Removing the optional flag of `title` and `desc` would prevent older formatted `[map]` bbcode to render.
image

Example above: 
  - first with `title` and `desc` does render.
  - the second without does not render.

That's a rather inconvenient regression.

Comment (brian-guru) [reply src/Providers/BBCodesProvider.php]: It sounds strange. It works well from my website: https://vietsuquan.com/d/1-chua-phu-xa-phuc-hoa-tu
I added 2 instances of bbcode map with and without popup content.

Comment (JeromeGillard) [reply src/Providers/BBCodesProvider.php]: Thanks for your feedback. Something might be wrong with my dev instance. I'll look into it and keep hou updated here.

File: js/src/forum/components/mapBBCode.js (hunk 2 above)
Comment (JeromeGillard) [reply js/src/forum/components/mapBBCode.js]: Ok, got it working on my end! I restored your default line, with translations included.

Somehow, in `BBCodesProvider.php` removing the `?` in the `div` part but not in the `[map]` solved the error I was having.

File: src/Providers/BBCodesProvider.php (hunk 1 above)
Comment (JeromeGillard) [reply src/Providers/BBCodesProvider.php]: Somehow, in `BBCodesProvider.php` removing the `?` in the div part but not in the  [map]` solved the error I was having.

**Comment by JeromeGillard** [comment]
Seems fine on my end. 

I've made the marker to be added only if `title` or `desc` are provided. Keeping the option that if none are, the map is displayed without marker.

@bientran Could you test your `main` branch on your end?
Thanks again for your contribution! 🎉

**Pull request merged**